lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -0,0 +1,209 @@
+		if (!metadata.gav) {
+			throw new Error(
+				"Metadata is missing GAV (group, artifact and version) " +
+					"information. This might indicate an unsupported SNAPSHOT version."
+			);
+		}
+		const gav = metadata.gav.split(":");
+		let pkgName = metadata.npmPackageName;
+		if (!this._sources) {
+			pkgName += "-prebuilt";
+		}
+
+		return {
+			metadata: Promise.resolve({
+				id: pkgName,
+				version: metadata.version,
+				dependencies: metadata.dependencies,
+				optionalDependencies: metadata.optionalDependencies,
+			}),
+			// Trigger installation of package
+			install: this._installer.installPackage({
+				pkgName,
+				groupId: gav[0],
+				artifactId: gav[1],
+				version: metadata.version,
+				classifier: this._sources ? "npm-sources" : null,
+				extension: this._sources ? "zip" : "jar",
+			}),
+		};
+	}
+	static async fetchAllVersions({ui5HomeDir, cwd} = {}) {
+		const installer = new Installer({
+			cwd: cwd ? path.resolve(cwd) : process.cwd(),
+			ui5HomeDir: path.resolve(
+				ui5HomeDir || path.join(os.homedir(), ".ui5")
+			),
+			snapshotEndpointUrlCb: this._snapshotEndpointUrlCb,
+		});
+		return await installer.fetchPackageVersions({
+			groupId: DIST_GROUP_ID,
+			artifactId: DIST_ARTIFACT_ID,
+		});
+	}
+	/**
+	 * Tries to detect whether ~/.m2/settings.xml exist, and if so, whether
+	 * the snapshot.build URL is extracted from there
+	 *
+	 * @param {string} [settingsXML=~/.m2/settings.xml] Path to the settings.xml.
+	 * 				If not provided, the default location is used
+	 * @param {boolean} [skipConfirmation=false] Whether to skip manual user
+	 * 				confirmation in the consle and enforce reading from settings.xml
+	 * @returns {Promise<string>} The resolved snapshot.build URL from ~/.m2/settings.xml
+	 */
+	static async _resolveSnapshotEndpointUrl(settingsXML, skipConfirmation) {
+		if (!skipConfirmation) {
+			const {default: yesno} = await import("yesno");
+			const ok = await yesno({
+				question:
+					"SnapshotURL is missing. " +
+					"Would you like to try to resolve it from ~/.m2/settings.xml? (yes)",
+				defaultValue: true,
+			});
+
+			if (!ok) {
+				throw new Error("SnapshotURL not resolved");
+			}
+		}
+
+		settingsXML =
+			settingsXML || path.resolve(path.join(os.homedir(), ".m2", "settings.xml"));
+		const {default: fs} = await import("graceful-fs");
+		const {promisify} = await import("node:util");
+		const readFile = promisify(fs.readFile);
+		const xml2js = await import("xml2js");
+		const parser = new xml2js.Parser({
+			preserveChildrenOrder: true,
+			xmlns: true,
+		});
+		let xmlContents;
+
+		try {
+			const fileContent = await readFile(settingsXML);
+			xmlContents = await parser.parseStringPromise(fileContent);
+		} catch (err) {
+			if (err.code === "ENOENT") {
+				// "File or directory does not exist"
+				throw new Error(`SnapshotURL not resolved. Settings.xml could not be found in ${settingsXML}`);
+			} else {
+				throw err;
```

Comment:
Consider adding some context to the error message:

```suggestion
				throw new Error(
					`Failed to read Maven configuration file from ${settingsXML}: ${err.message}`, {
					cause: err
				});
```